[WebGPU Swift] http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet.html does not pass

mwyrzykowski opened the pull request:
#### 1e997054c1b82f9b72d5cbc4a3218ed54c4af65d
<pre>
[WebGPU Swift] http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet.html does not pass
<a href="https://bugs.webkit.org/show_bug.cgi?id=300954">https://bugs.webkit.org/show_bug.cgi?id=300954</a>
<a href="https://rdar.apple.com/162832438">rdar://162832438</a>

Reviewed by Tadeu Zagallo.

Validation logic in resolveQuerySet was incorrect.

1:1 copy the Objective-C++ logic.

* Source/WebGPU/WebGPU/CommandEncoder.swift:

Canonical link: <a href="https://commits.webkit.org/301733@main">https://commits.webkit.org/301733@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6385e3ed39f0004458a1a2f2f81eba06ae3f5c20

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126839 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46480 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37457 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133845 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78447 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/108d3ade-4840-4ae2-a4bc-2b9b125e9848) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47105 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55015 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96525 "45 flakes 44 failures") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/feec7179-fc37-48d4-b6a3-f282e8ab3e41) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129787 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37699 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113457 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77039 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/5291395d-d888-42f4-9ea6-296c96e0d383) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36571 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77239 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107529 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31939 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136371 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53511 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41189 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105035 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54007 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109816 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104731 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26717 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50236 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28580 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50981 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53440 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59288 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52694 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56029 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54443 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on Source/WebGPU/WebGPU/CommandEncoder.swift:
```diff
@@ -1854,15 +1854,26 @@ extension WebGPU.CommandEncoder {
         }
-        guard self.validateResolveQuerySet(querySet: querySet, firstQuery: firstQuery, queryCount: queryCount, destination: destination, destinationOffset: destinationOffset) else {
+
+        guard self.validateResolveQuerySet
```
Patch looks good, but looks like the Swift compiler is unhappy with the formatting here

mwyrzykowski replied:
ah I changed that for the linter and now I see it no longer compiles :(

mwyrzykowski replied:
ok patch updated, just moved the opening `(` to the same line as `validateResolveQuerySet`

mwyrzykowski committed:
[WebGPU Swift] http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet.html does not pass
https://bugs.webkit.org/show_bug.cgi?id=300954
rdar://162832438

Reviewed by Tadeu Zagallo.

Validation logic in resolveQuerySet was incorrect.

1:1 copy the Objective-C++ logic.

* Source/WebGPU/WebGPU/CommandEncoder.swift:

Canonical link: https://commits.webkit.org/301733@main
